### PR TITLE
fixes #26489 - graphql: create host mutation

### PR DESCRIPTION
**Pull request by timogoebel**
This commit allows to create a host with a graphql mutation. 🎱

This needs https://github.com/theforeman/foreman/pull/6656.

### CODE REVIEW

**Comment by timogoebel** [comment]
@xprazak2: I hope #6656 is close to a merge. Mind taking a look?

**Comment by timogoebel** [comment]
Rebased.
@xprazak2: This is ready for a review. :-)

File: app/graphql/types/compute_profile.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module Types
+  class ComputeProfile < BaseObject
```
Comment (xprazak2) [inline app/graphql/types/compute_profile.rb:2]: Could you add queries as well please? It should be just 2 lines in `Types::Query`...

File: test/graphql/mutations/hosts/create_mutation_test.rb [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+          domainId: domain_id,
+          ownerId: owner_id,
+          rootPass: root_pass,
+          build: true
+        }
+      end
+      let(:variables) do
+        base_variables.merge(
+          mac: mac,
+          ownerId: owner_id
```
Comment (xprazak2) [inline test/graphql/mutations/hosts/create_mutation_test.rb]: Nit: `owner_id` is already in `base_variables` (on line 45)

**Comment by xprazak2** [comment]
Left just 2 comments, otherwise working well. Test failures look relevant.

File: app/graphql/types/compute_profile.rb (hunk 1 above)
Comment (timogoebel) [reply app/graphql/types/compute_profile.rb:2]: True, but I'd like to add tests for the queries as well. If you don't mind, let's do it in a follow up PR.

**Comment by timogoebel** [comment]
@xprazak2: Updated.

File: app/graphql/types/compute_profile.rb (hunk 1 above)
Comment (xprazak2) [reply app/graphql/types/compute_profile.rb:2]: Makes sense.

**Review by xprazak2 — APPROVED** [review]

**Comment by xprazak2** [comment]
Thanks!

File: app/graphql/types/compute_profile.rb (hunk 1 above)
Comment (timogoebel) [reply app/graphql/types/compute_profile.rb:2]: https://projects.theforeman.org/issues/27063

**Pull request merged**